api/src/test/java/com/epam/pipeline/acl/cluster/ClusterApiServiceTest.java

```diff
@@ -0,0 +1,451 @@
+        doReturn(nodeInstance).when(mockNodesManager).terminateNode(nodeInstance.getName());
+
+        assertThat(clusterApiService.terminateNode(nodeInstance.getName())).isEqualTo(nodeInstance);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldTerminateNodeWhenPermissionIsGranted() {
+        final NodeInstance nodeInstance = NodeCreatorUtils.getNodeInstance(1L, TEST_STRING);
+        doReturn(SIMPLE_USER).when(mockAuthManager).getAuthorizedUser();
+        initAclEntity(nodeInstance, AclPermission.READ);
+        doReturn(nodeInstance).when(mockNodesManager).terminateNode(nodeInstance.getName());
+        doReturn(nodeInstance).when(mockNodesManager).getNode(nodeInstance.getName());
+        doReturn(pipelineRun).when(mockPipelineRunManager).loadPipelineRun(eq(pipelineRun.getId()));
+
+        final NodeInstance returnedNodeInstance = clusterApiService.terminateNode(nodeInstance.getName());
+
+        assertThat(returnedNodeInstance).isEqualTo(nodeInstance);
+        assertThat(returnedNodeInstance.getMask()).isEqualTo(AclPermission.READ.getMask());
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDenyAccessToNodeTerminationWhenPermissionIsNotGranted() {
+        initAclEntity(nodeInstance);
+        doReturn(nodeInstance).when(mockNodesManager).terminateNode(nodeInstance.getName());
+        doReturn(nodeInstance).when(mockNodesManager).getNode(nodeInstance.getName());
+        doReturn(pipelineRun).when(mockPipelineRunManager).loadPipelineRun(eq(pipelineRun.getId()));
+
+        assertThrows(AccessDeniedException.class,
+                () -> clusterApiService.terminateNode(nodeInstance.getName()));
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldReturnStatsForNodeForAdmin() {
+        final List<MonitoringStats> statsList = Collections.singletonList(monitoringStats);
+        doReturn(statsList).when(mockUsageMonitoringManager).getStatsForNode(
+                nodeInstance.getName(), LocalDateTime.MIN, LocalDateTime.MAX);
+
+        final List<MonitoringStats> returnedStatsList =
+                clusterApiService.getStatsForNode(nodeInstance.getName(), LocalDateTime.MIN, LocalDateTime.MAX);
+
+        assertThat(returnedStatsList).hasSize(1).contains(monitoringStats);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldReturnStatsWhenPermissionIsGranted() {
+        doReturn(SIMPLE_USER).when(mockAuthManager).getAuthorizedUser();
+        initAclEntity(nodeInstance, AclPermission.READ);
+        doReturn(statsList).when(mockUsageMonitoringManager).getStatsForNode(
+                nodeInstance.getName(), LocalDateTime.MIN, LocalDateTime.MAX);
+        doReturn(nodeInstance).when(mockNodesManager).getNode(nodeInstance.getName(), filterPodsRequest);
+        doReturn(nodeInstance).when(mockNodesManager).getNode(nodeInstance.getName());
+        doReturn(pipelineRun).when(mockPipelineRunManager).loadPipelineRun(eq(pipelineRun.getId()));
```

Comment:
I wonder if we can extract all this duplicated mocking calls into some methods like the following:
```java
mockUser();
mockStats();
mockNodeInstance();
mockPipelineRun();
```